di/src/main/java/com/interface21/beans/factory/support/DefaultListableBeanFactory.java

```diff
@@ -58,30 +59,58 @@ public int getBeanDefinitionCount() {
     public <T> T getBean(final Class<T> clazz) {
         Object bean = beanRegistry.getBean(clazz);
         if (bean != null) {
-            return (T) bean;
+            return clazz.cast(bean);
         }
-
-        return (T) instantiateClass(clazz);
+        return clazz.cast(instantiateClass(clazz));
     }
 
     @Override
     public void clear() {}
 
+    @Override
+    public Object[] registerArgumentValues(Class<?> type, Class<?>[] parameterTypes) {
+        Object[] args = Arrays.stream(parameterTypes).map(this::getBean).toArray(Object[]::new);
+        return constructorArgumentValueHolder.addValueHolder(type, args);
+    }
+
+    private void initializeBeans() {
+        beanDefinitionMap
+                .values()
+                .forEach(
+                        beanDefinition -> {
+                            Object instance = createBean((AnnotationBeanDefinition) beanDefinition);
+                            beanRegistry.registeredBean(instance);
+                        });
```

Comment:
```suggestion
        beanDefinitionMap
                .values()
                .map(beanDefinition -> createBean((AnnotationBeanDefinition) beanDefinition))
                .forEach(instance -> beanRegistry.registeredBean(instance));
```

이렇게 해도 indent가 줄어서 좋을 것 같네요!